File: frontend/src/api/tasks.ts
import client from './client'
import type { TaskResponse } from '../types'

export const tasksApi = {
  getStatus: async <T>(task_id: string): Promise<TaskResponse<T>> => {
    const { data } = await client.get<TaskResponse<T>>(`/api/tasks/${task_id}`)
    return data
  },
}

File: api/routers/tasks.py
from __future__ import annotations

from fastapi import APIRouter

from api.dependencies import get_task
from api.schemas.tasks import TaskResponse

router = APIRouter()


@router.get("/{task_id}", response_model=TaskResponse)
async def get_task_status(task_id: str):
    """Poll the status of a background task."""
    return get_task(task_id)
